tests/unit/aiplatform/test_automl_text_training_jobs.py

```diff
@@ -0,0 +1,532 @@
+            multi_label=True,
+        )
+
+        model_from_job = job.run(
+            dataset=mock_dataset_text,
+            training_fraction_split=_TEST_FRACTION_SPLIT_TRAINING,
+            validation_fraction_split=_TEST_FRACTION_SPLIT_VALIDATION,
+            test_fraction_split=_TEST_FRACTION_SPLIT_TEST,
+            model_display_name=None,  # Omit model_display_name
+            sync=sync,
+        )
+
+        if not sync:
+            model_from_job.wait()
+
+        true_fraction_split = gca_training_pipeline.FractionSplit(
+            training_fraction=_TEST_FRACTION_SPLIT_TRAINING,
+            validation_fraction=_TEST_FRACTION_SPLIT_VALIDATION,
+            test_fraction=_TEST_FRACTION_SPLIT_TEST,
+        )
+
+        # Test that if defaults to the job display name
+        true_managed_model = gca_model.Model(display_name=_TEST_DISPLAY_NAME)
+
+        true_input_data_config = gca_training_pipeline.InputDataConfig(
+            fraction_split=true_fraction_split, dataset_id=mock_dataset_text.name,
+        )
+
+        true_training_pipeline = gca_training_pipeline.TrainingPipeline(
+            display_name=_TEST_DISPLAY_NAME,
+            training_task_definition=schema.training_job.definition.automl_text_classification_task,
+            training_task_inputs=_TEST_TRAINING_TASK_INPUTS_CLASSIFICATION,
+            model_to_upload=true_managed_model,
+            input_data_config=true_input_data_config,
+        )
+
+        mock_pipeline_service_create.assert_called_once_with(
+            parent=initializer.global_config.common_location_path(),
+            training_pipeline=true_training_pipeline,
+        )
+
+    @pytest.mark.parametrize("sync", [True, False])
+    def test_run_called_twice_raises(
+        self,
+        mock_pipeline_service_create,
+        mock_dataset_text,
+        mock_model_service_get,
+        sync,
+    ):
+        aiplatform.init(project=_TEST_PROJECT)
+
+        job = training_jobs.AutoMLTextTrainingJob(
+            display_name=_TEST_DISPLAY_NAME,
+            prediction_type="classification",
+            multi_label=True,
+        )
+
+        job.run(
+            dataset=mock_dataset_text,
+            model_display_name=_TEST_MODEL_DISPLAY_NAME,
+            training_fraction_split=_TEST_FRACTION_SPLIT_TRAINING,
+            validation_fraction_split=_TEST_FRACTION_SPLIT_VALIDATION,
+            test_fraction_split=_TEST_FRACTION_SPLIT_TEST,
+            sync=sync,
+        )
+
+        with pytest.raises(RuntimeError):
+            job.run(
+                dataset=mock_dataset_text,
+                model_display_name=_TEST_MODEL_DISPLAY_NAME,
+                training_fraction_split=_TEST_FRACTION_SPLIT_TRAINING,
+                validation_fraction_split=_TEST_FRACTION_SPLIT_VALIDATION,
+                test_fraction_split=_TEST_FRACTION_SPLIT_TEST,
+                sync=sync,
+            )
+
+    @pytest.mark.parametrize("sync", [True, False])
```

Comment:
@sasha-gitg Perhaps remove this as it doesn't add code coverage

Reply:
Does it add branch coverage?

Reply:
You know, actually it tests for this one line, so let's keep it:

```
if self._has_run:
            raise RuntimeError("AutoML Text Training has already run.")
```